controllers/operatorpolicy_status.go

```diff
@@ -273,10 +300,30 @@ func (r *OperatorPolicyReconciler) emitComplianceEvent(
 	return r.Create(ctx, event)
 }
 
+func (r *OperatorPolicyReconciler) removeObjsOfType(
+	ctx context.Context,
+	policy *policyv1beta1.OperatorPolicy,
+	gvk schema.GroupVersionKind,
+) error {
+	newRelObjs := make([]policyv1.RelatedObject, 0)
+
+	for _, relObj := range policy.Status.RelatedObjects {
+		if relObj.Object.Kind != gvk.Kind {
+			newRelObjs = append(newRelObjs, relObj)
+		}
+	}
+
+	policy.Status.RelatedObjects = newRelObjs
+
+	return r.Status().Update(ctx, policy)
```

Comment:
This whole function is not needed now, but it's worth mentioning that if it was used alongside `updateStatus`, this would result in 2 API calls, which would not be optimal. I would've preferred it to be a method on the policy status, and the caller would be expected to do this update call themselves.